docs/tutorials/Canopy/soil_plant_hydrology_tutorial.jl

```diff
@@ -0,0 +1,457 @@
+    return T(1.0 / rooting_depth) * exp(z / T(rooting_depth)) # 1/m
+end
+
+ψ63 = FT(-4 / 0.0098)
+Weibull_param = FT(4)
+a = FT(0.05 * 0.0098)
+
+conductivity_model =
+    PlantHydraulics.Weibull{FT}(K_sat_plant, ψ63, Weibull_param)
+
+retention_model = PlantHydraulics.LinearRetentionCurve{FT}(a)
+
+plant_ν = FT(0.7)
+plant_S_s = FT(1e-2 * 0.0098)
+
+plant_hydraulics_ps = PlantHydraulics.PlantHydraulicsParameters(;
+    area_index = area_index,
+    ν = plant_ν,
+    S_s = plant_S_s,
+    root_distribution = root_distribution,
+    conductivity_model = conductivity_model,
+    retention_model = retention_model,
+)
+
+n_stem = Int64(1)
+n_leaf = Int64(1)
+h_stem = FT(9)
+h_leaf = FT(9.5)
+compartment_midpoints = [h_stem / 2, h_stem + h_leaf / 2]
+compartment_surfaces = [zmax, h_stem, h_stem + h_leaf]
+
+plant_hydraulics_args = (
+    parameters = plant_hydraulics_ps,
+    n_stem = n_stem,
+    n_leaf = n_leaf,
+    compartment_midpoints = compartment_midpoints,
+    compartment_surfaces = compartment_surfaces,
+);
+
+# We may now collect all of the canopy component argument tuples into one 
+# arguments tuple for the canopy component models.
+
+canopy_component_args = (;
+    radiative_transfer = radiative_transfer_args,
+    photosynthesis = photosynthesis_args,
+    conductance = conductance_args,
+    hydraulics = plant_hydraulics_args,
+);
+
+# We also need to provide the shared parameter struct to the canopy as usual and
+# specify the domain for the canopy.
+
+z0_m = FT(2)
+z0_b = FT(0.2)
+
+shared_params = SharedCanopyParameters{FT, typeof(earth_param_set)}(
+    LAI,
+    h_stem + h_leaf,
+    z0_m,
+    z0_b,
+    earth_param_set,
+)
+
+canopy_model_args = (; parameters = shared_params, domain = land_domain.surface);
+
+# - We will be using prescribed atmospheric and radiative drivers from the 
+#   US-MOz tower, which we read in here. We are using prescribed
+#   atmospheric and radiative flux conditions, but it is also possible to couple
+#   the simulation with atmospheric and radiative flux models.
+
+include(
+    joinpath(
+        pkgdir(ClimaLSM),
+        "experiments/integrated/ozark/ozark_met_drivers_FLUXNET.jl",
+    ),
+);
+
+# We may now instantiate the integrated plant hydraulics and soil model. The 
+# Default diagnostics is transpiration. Here we will work with prescribed 
+# atmospheric and radiative flux conditions from the observations at the Ozark 
+# site as was done in the previous tutorial.
+
+land_input =
+    (transpiration = transpiration, atmos = atmos, radiation = radiation)
+
+land = SoilPlantHydrologyModel{FT}(;
+    land_args = land_input,
+    soil_model_type = soil_model_type,
+    soil_args = soil_args,
+    canopy_component_types = canopy_component_types,
+    canopy_component_args = canopy_component_args,
+    canopy_model_args = canopy_model_args,
+);
+
+# Now we can initialize the state vectors and model coordinates, and initialize 
+# the explicit/implicit tendencies as usual.
```

Comment:
Perhaps note that time-stepping Richards equation in the soil is done via an implicit method, while the canopy model can be stepped explicitly, so we use an IMEX (implicit-explicit) scheme for the combined model.